js.go

```diff
@@ -1844,7 +1844,14 @@ func (sub *Subscription) activityCheck() {
 	}
 
 	active := jsi.active
-	jsi.hbc.Reset(jsi.hbi)
+	// the timer could still have gone off even though it was stopped in removeSub,
```

Comment:
Would revert the changes here and instead few lines above change the condition to:
```
if jsi == nil || sub.closed {
    sub.mu.Unlock()
    return
}
```
(or maybe even `if sub.closed` would be enough)

Reply:
Yup.. that would work

Reply:
Like I said, this change can be reverted. Since jsi.hbc is set to nil at the same time (and under sub.mu.Lock) we set sub.closed = true in removeSub, we are ok with original code.